apps/demos/src/side-nav.component.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * @license
+ * Copyright 2019 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AfterContentInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  Input,
+  OnDestroy,
+  ViewEncapsulation,
+} from '@angular/core';
+import { NavigationEnd, Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+
+interface ComponentItem {
+  name: string;
+  examples: Array<{ name: string; route: string }>;
+}
+
+@Component({
+  selector: 'dt-demos-side-nav',
+  templateUrl: 'side-nav.component.html',
+  styleUrls: ['side-nav.component.scss'],
+  host: {
+    class: 'dt-demos-side-nav',
+  },
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DtDemosSideNav implements AfterContentInit, OnDestroy {
+  @Input()
+  get componentItems(): ComponentItem[] {
+    return this._componentItems;
+  }
+  set componentItems(values: ComponentItem[]) {
+    this._componentItems = values;
+    this._updateFilteredComponentItems();
+  }
+
+  @Input('componentItemValue')
+  get componentItemsFilterValue(): string {
+    return this._componentItemsFilterValue;
+  }
+  set componentItemsFilterValue(value: string) {
+    const filterValue = value.trim();
+
+    if (this._componentItemsFilterValue !== filterValue) {
+      this._componentItemsFilterValue = filterValue;
+      this._updateFilteredComponentItems();
+    }
+  }
+
+  get filteredComponentItems(): ComponentItem[] {
+    return this._filteredComponentItems;
+  }
+
+  private _selectedComponentName = '';
+  private _componentItemsFilterValue = '';
+  private _urlSubscription: Subscription;
+  private _componentItems: ComponentItem[] = [];
+  private _filteredComponentItems: ComponentItem[] = [];
+
+  constructor(
+    private readonly _router: Router,
+    private readonly _changeDetectorRef: ChangeDetectorRef,
+  ) {}
+
+  ngAfterContentInit(): void {
+    this._filteredComponentItems = [...this._componentItems];
+    this._urlSubscription = this._router.events.subscribe(event => {
```

Comment:
please unsubscribe

Reply:
Not going to touch the actual code in this PR if not needed.
Side nav is also going to live the whole lifespan of the app